File: python/1232_Check_If_It_Is_a_Straight_Line.py
class Solution:
    def checkStraightLine(self, c: List[List[int]]) -> bool:
        dx = c[1][0] - c[0][0]
        dy = c[1][1] - c[0][1]
        for i in range(2, len(c)):
            cdx = c[i][0] - c[0][0]
            cdy = c[i][1] - c[0][1]
            if cdx * dy != cdy * dx:
                return False
        return True

File: cpp/1232_Check_If_It_Is_a_Straight_Line.cpp
#include <vector>

using namespace std;

class Solution {
public:
    bool checkStraightLine(vector<vector<int>>& c) {
        int dx = c[1][0] - c[0][0];
        int dy = c[1][1] - c[0][1];
        for (int i = 2; i < c.size(); i++) {
            int cdx = c[i][0] - c[0][0];
            int cdy = c[i][1] - c[0][1];
            if (cdx * dy != cdy * dx) {
                return false;
            }
        }
        return true;
    }
};